test_runner/fixtures/pageserver/http.py

```diff
@@ -831,7 +824,7 @@ def timeline_spawn_download_remote_layers(
 
     def timeline_poll_download_remote_layers_status(
         self,
-        tenant_id: Union[TenantId, TenantShardId],
+        tenant_id: TenantId | TenantShardId,
```

Comment:
maybe you can define an additional type, that equals to `TenantId | TenantShardId`, rather than replacing it everywhere?

Reply:
Generally the clean end state would be to use shard IDs everywhere.  Either way, refactoring this is orthogonal to this PR.